addons/web/static/src/model/relational_model/record.js

```diff
@@ -874,7 +875,7 @@ export class Record extends DataPoint {
             }
             throw e;
         }
-        if (creation && records.length) {
+        if ((creation || nextID) && records.length) {
```

Comment:
for the nextId case, you need the records.length check (to throw the FetchRecordError)
In other words, records[0].id might crash